packages/protocol/socket.ts

```diff
@@ -290,13 +290,38 @@ export function addEventListener<M extends EventMethods>(
   }
 }
 
-// TODO This method should have the same signature as addEventListener.
-// It should only remove a specific event handler.
-export function removeEventListener<M extends EventMethods>(event: M) {
-  gEventListeners.delete(event);
+export function removeEventListener<M extends EventMethods>(
+  event: M,
+  handler?: (params: EventParams<M>) => void
+) {
+  switch (event) {
+    case "Analysis.analysisPoints":
+    case "Analysis.analysisResult": {
+      gEventListeners.delete(event);
+      break;
+    }
```

Comment:
I'm kind of afraid to touch the special casing around analysis stuff because I haven't dug in enough to understand it, so for now I'm leaving its behavior as-is.